crates/opencascade/src/primitives.rs

```diff
@@ -425,6 +429,21 @@ impl Face {
         Shape { inner }
     }
 
+    pub fn revolve(&self, origin: DVec3, axis: DVec3, angle: Angle) -> Solid {
```

Comment:
I'd like to suggest we make two functions

```
revolve((&self, origin: DVec3, axis: DVec3)
```

and

```
revolve_angle((&self, origin: DVec3, axis: DVec3, angle: Angle)
```

Full revolves feel like they're done often enough that it would be annoying to always have to specify the angle.

Reply:
I dont like have two functions for similar things. May be better turn angle parameter to be Option.

Reply:
Fair enough, the `Option<Angle>` API looks good.

I _might_ tweak it later as I observe usage. If there's a lot of `shape.revolve(origin, axis, None)` everywhere I might split out the angled version to a separate function. Nothing to worry about for this PR though.